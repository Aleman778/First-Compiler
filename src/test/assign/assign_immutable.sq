// ignore

// Test assign a new value to immutable variable.

fn main() {
    let a: i32 = 5;
    a = a + a * 2; //~ error: cannot assign twice to immutable variable
}
