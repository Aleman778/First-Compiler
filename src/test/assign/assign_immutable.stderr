error: cannot assign twice to immutable variable `a`
  --> $DIR/assign_immutable.sq:6:5
   |
LL |     let a: i32 = 5;
LL |     a = a + a * 2; //~ error: cannot assign twice to immutable variable
   |     ^^^^^^^^^^^^^ cannot assign twice to immutable variable